tensorrt_llm/_torch/cute_dsl_kernels/blackwell/kimi_k3_kda/fused_k123.py

```
@@ -0,0 +1,2306 @@
+
+"""
+Persistent Fused K1+K2+K3 Kernel for KDA.
+
+Fuses gate activation + cumsum + scaling (K1), intra sub-chunk Aqk/Akk (K2),
+and inter sub-chunk solve + merged inverse (K3) into a single persistent kernel.
+
+Grid: (NUM_SMS, 1, 1) — 148 persistent blocks, each loops over work units
+  Total work units = (NT/4) * H * B, distributed round-robin across SMs
+  Block i processes work units i, i+NUM_SMS, i+2*NUM_SMS, ...
+Block: 1024 threads (32 warps), warp-specialized with setmaxnreg (all groups 4-aligned):
+  Warps 0-15:  TMA+K1 fused (8×2, vec2, prefetch pipeline) – 4 WGs, 56 regs
```

Ruff (0.16.0)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)